src/coreclr/jit/compiler.h

```diff
@@ -994,13 +994,14 @@ class LclVarDsc
         regMaskTP regMask = RBM_NONE;
         if (GetRegNum() != REG_STK)
         {
-            if (varTypeUsesFloatReg(TypeGet()))
+            if (varTypeUsesIntReg(this))
             {
-                regMask = genRegMaskFloat(GetRegNum(), TypeGet());
+                regMask = genRegMask(GetRegNum());
             }
             else
             {
-                regMask = genRegMask(GetRegNum());
+                assert(varTypeUsesFloatReg(this));
```

Comment:
Is the reason you haven't added a check for kmask register is because we don't have a need currently and a check will be added here in future?

Reply:
Yes. When support for kmask is expanded beyond the 1 special intrinsic that is only introduced in lowering, then we'll want/need to update this and potential downstream handling as well.